kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/LocalStack.java

```diff
@@ -0,0 +1,187 @@
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.ServicePort;
+
+import io.kroxylicious.kms.provider.aws.kms.AwsKmsTestKmsFacade;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.executor.ExecResult;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+import io.kroxylicious.systemtests.resources.manager.ResourceManager;
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+import io.kroxylicious.systemtests.utils.TestUtils;
+import io.kroxylicious.systemtests.utils.VersionComparator;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.cmdKubeClient;
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * The type LocalStack.
+ */
+public class LocalStack implements AwsKmsClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocalStack.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+    public static final String LOCALSTACK_SERVICE_NAME = "localstack";
+    public static final String LOCALSTACK_DEFAULT_NAMESPACE = "localstack";
+    public static final String LOCALSTACK_HELM_REPOSITORY_URL = "https://localstack.github.io/helm-charts";
+    public static final String LOCALSTACK_HELM_REPOSITORY_NAME = "localstack";
+    public static final String LOCALSTACK_HELM_CHART_NAME = "localstack/localstack";
+    private static final String AWS_LOCAL_CMD = "awslocal";
+    private final String deploymentNamespace;
+    private String podName;
+    private String installedLocalStackVersion;
+    private String region;
+
+    /**
+     * Instantiates a new Aws.
+     *
+     */
+    public LocalStack() {
+        this.deploymentNamespace = LOCALSTACK_DEFAULT_NAMESPACE;
+    }
+
+    @Override
+    public String getAwsCmd() {
+        return AWS_LOCAL_CMD;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        boolean awsCloudSelected = Environment.AWS_USE_CLOUD.equalsIgnoreCase("true");
+        boolean keyIdDefaulted = Environment.AWS_ACCESS_KEY_ID.equals(Environment.AWS_ACCESS_KEY_ID_DEFAULT);
+        if (keyIdDefaulted) {
+            LOGGER.atInfo().log("Using AWS LocalStack");
+        }
+        else {
+            if (!awsCloudSelected) {
+                LOGGER.atWarn().log("AWS LocalStack selected, but AWS_ACCESS_KEY_ID is not defaulted. Please insert a correct key id.");
+            }
+        }
+        return keyIdDefaulted || !awsCloudSelected;
```

Comment:
We log a warning if `!awsCloudSelected == true` but then we also return true when we log that warning. I think thats contradictory.

I think, though I could be wrong, it should either be  
```suggestion
            if (!awsCloudSelected) {
                LOGGER.atWarn().log("AWS LocalStack selected, but AWS_ACCESS_KEY_ID is not defaulted. Please insert a correct key id.");
            }
        }
        return keyIdDefaulted && !awsCloudSelected;
```
or 
```suggestion
            if (!awsCloudSelected) {
                LOGGER.atWarn().log("AWS LocalStack selected, but AWS_ACCESS_KEY_ID is not defaulted. Please insert a correct key id.");
            }
        }
        return keyIdDefaulted || awsCloudSelected;
```

Thought the latter also sounds wrong to me because why would we return true for localstack available if `awsCloudSelected == true`.

Reply:
My intention here is avoid skipping both test cases and always run LocalStack unless the user specifies correctly the env vars `AWS_USE_CLOUD` and `AWS_ACCESS_KEY_ID`:

* `AWS_USE_CLOUD` == false, `AWS_ACCESS_KEY_ID` == test (default value) --> LocalStack
* `AWS_USE_CLOUD` == false, `AWS_ACCESS_KEY_ID` != test (non default value) --> LocalStack, but this config is wrong.
* `AWS_USE_CLOUD` == true, `AWS_ACCESS_KEY_ID` == test (default value) --> LocalStack, but this config is wrong
* `AWS_USE_CLOUD` == true, `AWS_ACCESS_KEY_ID` != test (non default value) --> AWS Cloud

Reply:
IMHO the value of `AWS_USE_CLOUD` should be the only thing that determines if we use cloud or localstack. Defaulting to localstack is clearly the right choice. 

We should never get here with `AWS_USE_CLOUD == true`. 

So this method could warn for 
`AWS_ACCESS_KEY_ID != test` but I don't think it needs to.

Reply:
> We should never get here with AWS_USE_CLOUD == true.

Oh wait the service loading means this doesn't hold true.

Reply:
I think this boils down to:

```
return dockerAvailable && !useCloud;
```